c++/src/kj/async-unix.c++

```diff
@@ -1051,7 +1051,7 @@ bool UnixEventPort::poll() {
         sigprocmask(SIG_SETMASK, &waitMask, &origMask);
         sigprocmask(SIG_SETMASK, &origMask, nullptr);
         break;
-#else
+#elif !defined(__EMSCRIPTEN__)
```

Comment:
Seems like it'd be more appropriate to #if-out the entire block relating to signals, rather than just this inner part. The work to set up `sigsetjmp()` above is redundant.